pkg/builder/status_request.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package builder
```

Comment:
Do you anticipate we'll add more builders over time? If so, maybe this organization would make more sense?
```
pkg/builders/
  statusrequest
  someothertype
```

and then in each package, we'd have a plain `Builder` type along with `NewBuilder`, e.g.

```go
package statusrequest

type Builder struct {}

func NewBuilder() *Builder {}
```

We could even align on a common function name to spit out the underlying object - perhaps `Build()`?

Reply:
Or if we're afraid that `pkg/builders/backup` would conflict with `var backup *v1.Backup`, we could suffix all builder packages with `builder` - `pkg/builders/backupbuilder` and then instead of saying `backupbuilder.NewBuilder()` it would just be `backupbuilder.New()`.

Reply:
I do expect we'd add more, yes; over time, I'd like this to replace the existing test obj helpers and be a little more succinct and standardized.

What's the motivation for having a subpackage per type?  I'm not sure on that aspect yet.

+1 for renaming to `Build()` to get the obj

Reply:
I was hoping to eliminate the duplication of "builder(s)" in `builders.NewStatusRequestBuilder`.

Reply:
maybe it should just go into `pkg/statusrequest`, as a `type Builder struct` and `NewBuilder()` constructor?  And over time as we relocate controller logic into separate packages, each of those can define their own?

Reply:
That seems reasonable. Although we'll have problems with single-word types, such as `backup` and `restore` (trying to use as both a variable name and package name).